org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/ClasspathParams.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.ls.core.internal;
+
+import org.eclipse.lsp4j.jsonrpc.validation.NonNull;
+import org.eclipse.xtext.xbase.lib.Pure;
+import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+
+/*
+ * Representation of a request that client send to server to resolve class path
+ */
+@SuppressWarnings("all")
+public class ClasspathParams {
+	/*
+	 *  the fully qualified name of startupClass
+	 */
+	@NonNull
+	private String startupClass;
```

Comment:
i tested locally, seems whether to serialize/deserialize a field depends on how you create the GsonBuilder. it does offer some extension method to create GsonBuilder to ignore fields not marked with expose, but lsp4j seems to use the default one